### DISCUSSION TITLE
🚀 Feature request - consistent sort order with duplicate values in Control Panel entries list

### DISCUSSION THREAD

**Discussion by swierczek** [ExpressionEngine/ExpressionEngine#3596] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
The Entry sort order at admin.php?/cp/publish/edit is inconsistent with duplicate values. In the following screenshots, entries 1-9 have the same date but the order they are actually returned is random with each page loads (likely due to MySQL's spec if the sort order isn't clearly defined).

<img width="653" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/63d5ac10-4f75-4bff-aca9-ab852488bccd">

After reloading the page once:
<img width="622" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/f6334ccb-f004-470e-b460-76515f874cb3">

After reloading the page again:
<img width="628" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/3d1ad99b-e350-4ca7-87f8-f23e95b89cd2">

For what it's worth, the channel entries tag seems to work how I'd expect:

```
{exp:channel:entries
	channel="test"
	orderby="date"
}
	{title}
{/exp:channel:entries}
```

Outputs 10...1

**Describe the solution you'd like**
I'd expect the order of results to behave essentially like `ORDER BY date, entry_id`. In this case, even though I'm sorting by `date` only, I'd expect to see the entries listed 10...1 because 9...1 all have the same `date` value.

**Describe alternatives you've considered**
None

**Teachability, Documentation, Adoption, Migration Strategy**
This would be a trivial update to the sort order with no additional teaching or documentation needed.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by swierczek** [comment]
After investigating a bit, it looks like `->order('entry_id', $table->sort_dir)` just needs to be added to `/system/ee/ExpressionEngine/Controller/Publish/Edit.php` on line ~170 to become:

```
$entries->order($sort_field, $table->sort_dir)
    ->order('entry_id', $table->sort_dir)
    ->limit($filter_values['perpage'])
    ->offset($offset);
$entries = $entries->all();
```

which will always sort secondarily by entry_id for consistent results:

<img width="689" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/3dc5b11d-a308-473a-8359-dc65d13c5b7e">

**Comment by intoeetive** [comment]
This makes a lot of sense! Thank you
https://github.com/ExpressionEngine/ExpressionEngine/pull/3597
